src/collection/cereal.rs

```diff
@@ -62,6 +66,28 @@ where
     Ok(map)
 }
 
+impl<'de> Deserialize<'de> for Template {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let value: Value = Deserialize::deserialize(deserializer)?;
+
+        let str_value =
+            match value {
+                Value::Bool(b) => b.to_string(),
+                Value::Number(n) => n.to_string(),
+                Value::String(s) => s,
+                Value::Null => "null".to_string(),
+                _ => return Err(Error::custom(
+                    "Invalid type, must be a string, number, boolean, or null",
+                )),
+            };
+
+        Template::try_from(str_value).map_err(|e| Error::custom(e.to_string()))
+    }
+}
+
```

Comment:
Could you add some tests for this please? I just wrote up a quick one, if you just add some more test cases to cover the other types to it I think that'll be good. If you want you can add a second test to cover error cases as well.

```rust
#[cfg(test)]
mod tests {
    use crate::template::Template;
    use rstest::rstest;
    use serde_test::{assert_de_tokens, Token};

    #[rstest]
    #[case(Token::Str("true"), "true")]
    #[case(Token::Bool(true), "true")]
    fn test_deserialize_template(#[case] token: Token, #[case] expected: &str) {
        assert_de_tokens(&Template::from(expected), &[token]);
    }
}
```